resolwe/flow/migration_ops.py

```diff
@@ -0,0 +1,351 @@
+import inspect
+
+import jsonschema
+
+from django.db.migrations.operations import base
+
+from resolwe.flow.models.utils import validation_schema
+from resolwe.flow.utils import dict_dot
+
+FIELD_SCHEMA = validation_schema('field')
+
+
+class DataDefaultOperation(object):
+    """Abstract data default generator."""
+
+    def prepare(self, data, from_state):
+        """Prepare data object migration.
+
+        Called after all relevant process schemas have already
+        been migrated.
+
+        :param data: Queryset containing all data objects that need
+            to be migrated
+        :param from_state: Database model state
+        """
+        raise NotImplementedError
+
+    def get_default_for(self, data, from_state):
+        """Return default for given data object.
+
+        :param data: Data object instance
+        :param from_state: Database model state
+        """
+        raise NotImplementedError
+
+
+class ConstantDataDefault(DataDefaultOperation):
+    """Data default, which generates constant values."""
+
+    def __init__(self, value):
+        self.value = value
+
+    def prepare(self, data, from_state):
+        """Prepare data object migration.
+
+        Called after all relevant process schemas have already
+        been migrated.
+
+        :param data: Queryset containing all data objects that need
+            to be migrated
+        :param from_state: Database model state
+        """
+        return
+
+    def get_default_for(self, data, from_state):
+        """Return default for given data object.
+
+        :param data: Data object instance
+        :param from_state: Database model state
+        """
+        return self.value
+
+
+class ResolweProcessOperation(base.Operation):
+    """
+    Abstract resolwe process migration operation.
+    """
+
+    reduces_to_sql = False
+    atomic = True
+    reversible = False
+
+    def __init__(self, process, schema_type, **kwargs):
+        """
+        Construct a process migration operation.
+
+        :param process: Process slug
+        :param schema_type: Type of schema (must be either ``input`` or
+            ``output``)
+        """
+
+        self.process = process
+        self.schema_type = schema_type
+        if self.schema_type not in ('input', 'output'):
+            raise ValueError("Schema type must be either input or output")
+
+        # Unfortunately, we can't easily get the identifier of the current
```

Comment:
I don't like this very much, but I don't see another way as there is no method that gets called with the current migration instance and Django's migration log is only updated after the migration has been applied.